# Assignment Week 1 (not graded)

## Prelude

In this exercise, you will do some basic operations using Python. This assignment is not graded.

## Exercise 1

Consider a rectangle with the small side of length 5 and the long side of length 8.

**1.1** Defining variables

Define variables `long_side` and `short_side` to represent the dimensions of the two sides.




In [1]:
long_side = 8 #Define long_side and short_side
short_side = 5

**1.2** Display the value of the long side.




In [2]:
print(long_side) #print long_side

8


**1.3** Which type does the variables `long_side` have?




In [3]:
print(f"long_side is of the type: {type(long_side).__name__}") #Print Information, then get the type of long_side and print the types name

long_side is of the type: int


You realise that your measurement was not precise — something we will later get to know
as measurement error. A more precise measurement reveals that the length of the long
side is 8.3.

**1.4** Overwriting variables

Overwrite the variable `long_side` defined above with 8.3. Which type does it have now? 




In [4]:
long_side = 8.3 #Overwrite long_side with precise value
print(f"long_side is now of the type: {type(long_side).__name__}") #Print Information, then get the type of long_side and print the types name

long_side is now of the type: float


**1.5** Operations with variables

Define the variable `perimeter_rectangle` containing the perimeter of the rectangle. Use operations on `long_side` and `short_side` when assigning `perimeter_rectangle`. Display its contents.




In [5]:
perimeter_rectangle = 2*(long_side + short_side) # U = 2(a+b)
print(perimeter_rectangle)

26.6


**1.6** More operations

Define the variable `area_rectangle` containing the area of the rectangle. Use operations on `long_side` and `short_side` when assigning to `area_rectangle`. Display its contents.




In [6]:
area_rectangle = long_side * short_side # A = a*b
print(area_rectangle)

41.5


**1.7** Which types does `perimeter_rectangle` and `area_rectangle` have?




In [7]:
print(f"perimeter_rectangle is of the type: {type(perimeter_rectangle).__name__} \narea_rectangle is of the type: {type(area_rectangle).__name__}") #Print Information, then get the type of long_side and print the types name

perimeter_rectangle is of the type: float 
area_rectangle is of the type: float


## Exercise 2

**2.1** Variable creation

Consider a square with side length 7. Assign this value to a variable `square_side`.




In [8]:
square_side = 7

**2.2** Perimater calculations

Calculate `perimeter_square` with obvious contents. In doing so, use the variable `square_side`, but only once. Display its contents.




In [9]:
perimeter_square = 4*square_side # U = 4*a
print(perimeter_square)

28


**2.3** Area calculations

Calculate `area_square` with obvious contents. In doing so, use the variable `square_side`, but only once.




In [10]:
area_square = square_side**2 # A = a^2

**2.4** Which types does `perimeter_square` and `area_square` have?




In [11]:
print(f"perimeter_square is of the type: {type(perimeter_square).__name__} \narea_squarw is of the type: {type(area_square).__name__}") #Print Information, then get the type of long_side and print the types name

perimeter_square is of the type: int 
area_squarw is of the type: int


## Exercise 3

Use the logical operators '>', '<', '==' to answer the following questions:

**3.1** Is the long side of the rectangle smaller than the side of the square?




In [12]:
print(long_side < square_side)

False


**3.2** Perimeter logic

Is the perimeter of the rectange larger or equal than the perimeter of the square?




In [13]:
print(perimeter_rectangle >= perimeter_square)

False


**3.3** Is the area of the rectangle equal to the area of the square?




In [14]:
print(area_rectangle == area_square)

False


**3.4** Definition of Boolean variables

Define a Boolean variable named `short_smaller_than_square`, which is true if the short side of the rectangle shorter than the side of the square and false otherwise




In [15]:
short_smaller_than_square = short_side < square_side

**3.5** Defining more Boolean variables

Define a Boolean variable named `long_smaller_than_square` which is true if long side of the rectangle is shorter than the side of the square and false otherwise




In [16]:
long_smaller_than_square = long_side < square_side

**3.6** Which types do the two comparison variables have?




In [17]:
print(f"long_smaller_than_square is of the type: {type(long_smaller_than_square).__name__} \nshort_smaller_than_square is of the type: {type(short_smaller_than_square).__name__}") #Print Information, then get the type of long_side and print the types name

long_smaller_than_square is of the type: bool 
short_smaller_than_square is of the type: bool


Use the variables `short_smaller_than_square` and `long_smaller_than_square` to answer
the questions below. Note: each cell should consist of one operation using Boolean operators.

**3.7** Are both sides of the rectangle shorter than the side of the square?




In [18]:
print(short_smaller_than_square == long_smaller_than_square)

False


**3.8** Or logic

Is any of the sides of the rectangle shorter than the side of the square?




In [19]:
print(short_smaller_than_square or long_smaller_than_square)

True


**3.9** And logic

Is it true that the short side of the rectangle is shorter than the side of the square AND the long side of the rectangle is at least as long as the side of the square?




## Exercise 4


In this exercise, you will practice producing clear and readable texts by improving the
structure of a convoluted and poorly written explanation.

The texts below are difficult to read and to follow. You will first identify their
structure. Next, you will break them down into clear chunks, and apply markdown
formatting. 

In particular, your task will be to:


1. **Analyze each text**: Identify different parts of each text, such as calculations,
  results, and logical comparisons. As you go through it, think about where paragraph
  breaks should be added, how to structure the information more clearly, and which parts
  of each text need emphasis.

2. **Reformulate each text**:

   * Break each text into paragraphs: Each paragraph should contain a chunk of logically
     connected information. Feel free to add headings.
   * Use numbered and bulleted lists: When explaining steps, results, or comparisons,
     break these down into lists so the reader can follow along easily. Use numbered
     lists for ordered arguments and bulleted lists for unordered arguments.
   * Use inline code: Whenever you refer to variables or operations, format them as
     inline code (e.g. `long_side`, `perimeter_rectangle`) to make very clear that you
     refer to code there.
   * Create clear connections between ideas: Make sure that each section logically
     follows the previous one, and break down long, complex sentences that overwhelm the
     reader. 
   * Do not re-write the texts beyond:
     - Adding headings or putting structure like lists.
     - Punctuation and connecting words (e.g., removing an "and" connecting two parts of
       a long sentence is fine when you break that sentence up or removing parts of a
       sentence like "We now describe X" when you turn "X" into a heading.).

**4.1** Description of procedures and results from the exercises above


This text describes what we did in the exercises above. Reformulate it along the lines
described above.

> In this assignment, we performed calculations and comparisons between a rectangle and
> a square. The rectangle's dimensions were as follows. The length of the rectangle's
> long side was initially defined as long_side = 8, but it was later updated to a more
> precise value of 8.3, and the length of its short side was defined as short_side = 5.
> We then obtained the rectangle's perimeter and area. The perimeter of the rectangle
> was calculated as:  perimeter_rectangle = 2 * (long_side + short_side) = 26.6 and the
> rectangle's area was calculated as: area_rectangle = long_side * short_side = 41.5.
> The square's dimensions follow. The side length of the square was defined as
> square_side = 7. Its perimeter and area are described now. The square's perimeter was
> calculated as: perimeter_square = 4 * square_side = 28 and the area of the square was
> calculated as: area_square = square_side^2 = 49. We performed various comparisons
> between the rectangle and square using both direct comparisons and boolean variables.
> First the side lengths. We checked whether the long side of the rectangle was smaller
> than the side of the square: long_side < square_side, this was false. Then the
> perimeter, where we checked whether the perimeter of the rectangle was greater than or
> equal to that of the square: perimeter_rectangle >= perimeter_square, this was false.
> Finally, for the areas we checked whether the areas of the rectangle and the square
> were equal: area_rectangle == area_square. This was false, too. Finally, we made more
> complex comparison using Boolean variables. We created two Boolean variables,
> short_smaller_than_square checks whether the rectangle's short side is shorter than
> the side of the square and long_smaller_than_square checks whether the rectangle's
> long side is shorter than the side of the square. We then made three comparisons.
> Their outcomes are in parentheses. First, whether the sides of the rectangle are
> smaller than the square's side (False); second, whether at least one side of the
> rectangle is smaller than the square's side (True); finally, whether the short side of
> the rectangle is smaller than the square's side and the long side is at least as long
> (True).

# Description of the exercises

 #### In this assignment, we performed calculations and comparisons between a rectangle and a square.


  ### The rectangle's dimensions were as follows.
 The length of the rectangle's long side was initially defined as `long_side = 8`.
 But it was later updated to a more
 precise value of **8.3**.
 And the length of its short side was defined as `short_side = 5`.


 ### We then obtained the rectangle's perimeter and area.
 The perimeter of the rectangle
 was calculated as:  `perimeter_rectangle = 2 * (long_side + short_side) = 26.6`.
  And the
 rectangle's area was calculated as: `area_rectangle = long_side * short_side = 41.5`.


 ### The square's dimensions follow.


 The side length of the square was defined as
 `square_side = 7`.


 ### Its perimeter and area are described now.
 **1.** The square's perimeter was
 calculated as: `perimeter_square = 4 * square_side = 28`.
 **2.** And the area of the square was
 calculated as: `area_square = square_side^2 = 49`.


 ### We performed various comparisons between the rectangle and square using both direct comparisons and boolean variables.
 + **First the side lengths.**
 We checked whether the long side of the rectangle was smaller
 than the side of the square: `long_side < square_side`, this was false.
 + **Then the perimeter.**
 Where we checked whether the perimeter of the rectangle was greater than or
 equal to that of the square:
 `perimeter_rectangle = perimeter_square`, this was false.
 + **Finally, for the areas**
 We checked whether the areas of the rectangle and the square
 were equal:
 `area_rectangle == area_square`. This was false, too.
 + **Finally, we made more complex comparison using Boolean variables.**
  We created two Boolean variables,
 `short_smaller_than_square` checks whether the rectangle's short side is shorter than
 the side of the square.
  And `long_smaller_than_square` checks whether the rectangle's
 long side is shorter than the side of the square.

 + **We then made three comparisons. Their outcomes are in parentheses.**
 **1.** First, whether the sides of the rectangle are
 smaller than the square's side (**False**).
 **2.** Second, whether at least one side of the
 rectangle is smaller than the square's side (**True**);
 **3.** Finally, whether the short side of
 the rectangle is smaller than the square's side and the long side is at least as long
 (**True**).

**4.2** Faulty conclusions in an article

Description of the flaws in the conclusions reached in the article about musicians' life expectancy.


In the survey, you read about the life expectancy of musicians playing different
genres. The article is reported below as a reminder:

> "Musicians from the older genres – blues, jazz (including bebop and dixieland),
> country (including country and western, boogie woogie, honky tonk and bluegrass), and
> gospel (including spiritual and Christian rock) – enjoyed, on average, similar
> lifespans as those from the US population with the same year of birth and gender.
> 
> The next group – R&B (including doo wop and soul), pop, folk (including ballad and
> polka) and world music – had lower life expectancies compared with the US population.
> 
> Thereafter, the gap between population lifespans and average age of death for the more
>  recent genres – rock (including rockabilly), electronic (including experimental,
> techno, disco, and funk), punk, metal, rap and hip hop – widens.
> 
> [...]
> 
> Many musicians from younger genres – rock, electronic, punk, metal, rap, and hip hop –
> appear unlikely to live long enough to acquire the illnesses of middle and old age."

This article misses some important alternative explanations for the seemingly lower life
expectancy of musicians playing newer genres. Of course, we do not expect you to come up
with the arguments why the analysis is deeply flawed yourself *at this point* (towards
the end of the course, you should have made some progress). However, the text below
describes some of the most important issues in the article. Please re-write it along the
lines described above.

The article compares two things, namely the average age at death of musicians across
different genres with the life expectancy of the US population for similar birth years
as the musicians of a particular genre. The two concepts are closely aligned when
comparing cohorts of individuals all of whom have died already (say, all people born in
1850s England) but cannot, however, calculate life expectancy in this way when some
individuals in question are still alive; hence, in order to deal with present-day
populations, we need different concepts of mortality. Population life expectancy in the
graph (likely) displays the so-called period life expectancy, which uses current
mortality rates (beyond 99.9% for 27-year-olds, closer to 80% for 90-somethings) across
all age groups to estimate the average number of years a hypothetical cohort would live
if those rates remained constant throughout their lifetime. For musicians, however, the
article uses average age at death and neglects all people who are still alive; a
phenomenon called right-censoring which implies that not many deaths of 80-year-old hip
hop musicians can be recorded because there are hardly any 80-year-old hip hop musicians
yet. So the age at death of musicians is mechanically linked to the age of a genre.

# Conclusion of the article

The article compares two things, namely the **average age at death of musicians** across
different genres with the **life expectancy of the US population** for similar birth years
as the musicians of a particular genre.

The two concepts are **closely aligned** when
comparing cohorts of individuals all of whom have died already (*say, all people born in
1850s England*) but **cannot**, however, calculate life expectancy in this way when some
individuals in question are still alive; hence, in order to deal with present-day
populations, **we need different concepts of mortality**.

Population life expectancy in the
graph (*likely*) displays the so-called **period life expectancy**, which uses **current
mortality rates** (*beyond 99.9% for 27-year-olds, closer to 80% for 90-somethings*) across
all age groups to **estimate the average number** of years a hypothetical cohort would live
if those rates remained constant throughout their lifetime.

For musicians, however, the
article uses **average age at death** and **neglects all people who are still alive**. A
phenomenon called **right-censoring** which implies that not many deaths of 80-year-old hip
hop musicians can be recorded because there are hardly any 80-year-old hip hop musicians
yet. So the **age at death** of musicians is mechanically linked to **the age of a genre**.

In [20]:
print(short_smaller_than_square and long_side >= square_side)

True
